doc/src/tools/kani.md

```diff
@@ -1,7 +1,124 @@
 # Kani Rust Verifier
 
 The Kani Rust Verifier is a bit-precise model checker for Rust.
-This page will give more details on how to use Kani to verify the standard library.
-You can find more informations about how to install and use Kani in the
-[Kani book](https://model-checking.github.io/kani/).
+Kani is designed to prove safety properties in your code as well as
+the absence of undefined behavior. It uses formal methods under the hood to ensure that
+Rust programs adhere to user specified properties.
+
+You can find more informations about how to install in [this section of the Kani book](https://model-checking.github.io/kani/install-guide.html).
+
+## Usage
+
+Consider a rust function that takes an integer and returns its absolute value and
+a kani proof that invokes the function that you want to verify
+
+``` rust
+// src/main.rs
+fn abs_diff(a: i32, b: i32) -> i32 {
+    if a > b {
+        a - b
+    } else {
+        b - a
+    }
+}
+
+#[cfg(kani)]
+#[kani::proof]
+fn harness() {
+    let a = kani::any::<i32>();
+    let b = kani::any::<i32>();
+    let result = abs_diff(a, b);
+    kani::assert(result >= 0, "Result should always be more than 0");}
+```
+
+Running the command `cargo kani` in your cargo crate will give the result
+
+```
+...Metadata about the Kani run
+
+RESULTS:
+Check 1: abs_diff.assertion.1
+         - Status: FAILURE
+         - Description: "attempt to subtract with overflow"
+         - Location: src/main.rs:5:9 in function abs_diff
+
+... Other properties that might have failed or succeeded.
+
+Summary:
+Verification failed for - harness
+Complete - 0 successfully verified harnesses, 1 failures, 1 total.
+```
+
+
+## Using Kani to verify the rust standard library
+
+To aid the std library verification effort, Kani provides a subcommand out of the box to help you get started.
+
+### Step 1
+
+Modify your local copy of the rust std library by writing proofs for the functions/methods that you want to verify.
+
+For example, insert this short blob into your copy of the std library. This blob imports the building-block APIs such as
+`assert`, `assume`, `proof` and function-contracts such as `proof_for_contract` and `fake_function`.
```

Comment:
Add a link about `function-contracts`, maybe to our RFC. We need a documentation for it.